lib/src/components/NxModal/__tests__/NxModal.test.tsx

```diff
@@ -33,17 +38,50 @@ describe('NxModal', function() {
 
   it('merges any passed in className to the nx-modal div', function() {
     const nxModal = shallow(
-      <NxModal className="test"/>
+      <NxModal className="test" onClose={dummyCloseHandler}/>
     );
     const nxModalDiv = nxModal.find('.nx-modal');
     expect(nxModalDiv.hasClass('test')).toEqual(true);
   });
 
   it('includes any passed in attributes to the nx-modal div', function() {
     const nxModal = shallow(
-      <NxModal id="modal-id" data-foo="bar"/>
+      <NxModal id="modal-id" data-foo="bar" onClose={dummyCloseHandler}/>
     );
     expect(nxModal.find('.nx-modal').prop('id')).toEqual('modal-id');
     expect(nxModal.find('.nx-modal').prop('data-foo')).toEqual('bar');
   });
+
+  describe('NxModal event listener support', () => {
+    let container: HTMLDivElement | null;
+
+    it('executes onClose method when pressing ESC key', function () {
+      const mockCallBack = jest.fn();
+      const nxModal = (
+        <div>
+          <NxModal id="modal-id" onClose={mockCallBack}/>
+        </div>
+      );
+
+      act(() => {
+        mount(nxModal, {attachTo: container});
+      });
+
+      document.dispatchEvent(new KeyboardEvent('keydown', {key: 'Escape'}));
+      expect(mockCallBack).toHaveBeenCalled();
```

Comment:
Prefer the matcher `toHaveBeenCalledWith()` for more control.

Reply:
> Prefer the matcher `toHaveBeenCalledWith()` for more control.

Since the function doesn't accept parameters (because it's the modalCloseHandler, not the actual listener that it's wired on the modal) I've used instead the `toHaveBeenNthCalledWith` to ensure it was only called once and with no parameters

Reply:
> > Prefer the matcher `toHaveBeenCalledWith()` for more control.
> 
> Since the function doesn't accept parameters (because it's the modalCloseHandler, not the actual listener that it's wired on the modal) I've used instead the `toHaveBeenNthCalledWith` to ensure it was only called once and with no parameters: [0965fe5](https://github.com/sonatype/sonatype-react-shared-components/commit/0965fe5f2debfca69a24523bb35fa7d5fffaa13f)

Prefer `toHaveBeenCalledTimes(1)` instead. 
`toHaveBeenNthCalledWith` is used to check the parameters supplied to the function on an nth call. It only works in your test because you're not supplying the fn with the rest of parameters.
See https://jestjs.io/docs/en/expect#tohavebeennthcalledwithnthcall-arg1-arg2-

Reply:
Changed validation of number of times called: d945d0a

Reply:
> > > Prefer the matcher `toHaveBeenCalledWith()` for more control.
> > 
> > 
> > Since the function doesn't accept parameters (because it's the modalCloseHandler, not the actual listener that it's wired on the modal) I've used instead the `toHaveBeenNthCalledWith` to ensure it was only called once and with no parameters: [0965fe5](https://github.com/sonatype/sonatype-react-shared-components/commit/0965fe5f2debfca69a24523bb35fa7d5fffaa13f)
> 
> Prefer `toHaveBeenCalledTimes(1)` instead.
> `toHaveBeenNthCalledWith` is used to check the parameters supplied to the function on an nth call. It only works in your test because you're not supplying the fn with the rest of parameters.
> See https://jestjs.io/docs/en/expect#tohavebeennthcalledwithnthcall-arg1-arg2-

Changed validation of number of times called: d945d0a